pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -246,14 +278,23 @@ func CreateAndInitSupervisordVolume(client lclient.Client) (string, error) {
 	}
 
 	supervisordLabels := GetSupervisordVolumeLabels()
-	_, err = client.CreateVolume(supervisordVolumeName, supervisordLabels)
+	supervisordVolumes, err := client.GetVolumesByLabel(supervisordLabels)
 	if err != nil {
-		return "", errors.Wrapf(err, "Unable to create supervisord volume for component")
+		return "", errors.Wrapf(err, "unable to retrieve supervisord volume for component")
+	}
+
+	if len(supervisordVolumes) == 0 {
+		_, err := client.CreateVolume(supervisordVolumeName, supervisordLabels)
+		if err != nil {
+			return "", errors.Wrapf(err, "unable to create supervisord volume for component")
+		}
+	} else {
+		supervisordVolumeName = supervisordVolumes[0].Name
```

Comment:
What happens if we overwrite the supervisord binary on the volume while another process is copying it to the container?

I guess, would it be better to have one supervisord volume per component, rather than for all of docker?

Reply:
I was trying to test this, I had the same concerns when writing this PR. I was trying to do simultaneous `odo push` at the same time. But honestly, I couldn't hit the issue.

But I guess we can move it a supervisord volume per odo component, to be safe!

Reply:
addressed in review commit